craft_application/commands/lifecycle.py

```diff
@@ -221,7 +222,7 @@ def _run(
             self._run_lifecycle(parsed_args, step_name)
         except Exception as err:
             if debug:
-                emit.progress(str(err), permanent=True)
+                handle_runtime_error(self._app, err)
```

Comment:
I think you should augment [this test](https://github.com/canonical/craft-application/blob/main/tests/unit/commands/test_lifecycle.py#L723) to ensure the error is being handled by the emitter

Reply:
Ideally the test would raise a craft-parts-like error and check that the error details were emitted (which is the behavior we ultimately want anyway)